zkevm-circuits/src/ecc_circuit/util.rs

```diff
@@ -2,8 +2,17 @@ use eth_types::Field;
 use halo2_base::{AssignedValue, QuantumCell};
 use halo2_ecc::{bigint::CRTInteger, ecc::EcPoint, fields::FieldExtPoint};
 
+// Total number of rows allowable for ECC circuit
 pub const LOG_TOTAL_NUM_ROWS: u32 = 20;
 
+// Cell usage accounting for EcAdd, EcMul and EcPairing
+// Roud up to nearest 100
+pub(super) const EC_ADD_CELLS: usize = 6_900;    // actual: 6_851
+pub(super) const EC_MUL_CELLS: usize = 405_500;  // actual: 405_476
+pub(super) const EC_PAIRING_CELLS: usize = 6_627_500; // actual: 6_627_442
```

Comment:
To observe actual cell usage, trace-level log of ctx object's cursor position is added to allow direct row accounting. To observe cell usage for ECC, run:
`RUST_BACKTRACE=1 RUST_LOG=trace cargo test -p zkevm-circuits --features test ecc_circuit::test::test_ecc_circuit`

> Sample Output:
```
...
> [ECC] ==> EcAdd Assignmnet START:
> Ctx cell pos: [(0, 0)]
> [ECC] EcAdd Inputs Assigned:
> Ctx cell pos: [(0, 1014)]
> [ECC] EcAdd Assignmnet END:
> Ctx cell pos: [(0, 6851)]
...
> [ECC] ==> EcMul Assignmnet START:
> Ctx cell pos: [(0, 342550)]
> [ECC] EcMul Inputs Assigned:
> Ctx cell pos: [(0, 343298)]
> [ECC] EcMul Assignmnet END:
> Ctx cell pos: [(0, 748027)]
...
> [ECC] ==> EcPairing Assignment START:
> Ctx cell pos: [(19, 693721)]
> [ECC] EcPairing g1s Assigned:
> Ctx cell pos: [(19, 695073)]
> [ECC] EcPairing g2s Assigned:
> Ctx cell pos: [(19, 697777)]
> [ECC] EcPairing Inputs RLC Assigned:
> Ctx cell pos: [(19, 700082)]
> [ECC] EcPairingAssignment END:
> Ctx cell pos: [(25, 1029779)]
...
```
A blinding area of 12 rows is enabled for each advise column within the test case.